sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -1420,6 +1420,22 @@ abstract class CSVSuite
     }
   }
 
+  test("SPARK-49125: write CSV files with duplicated field names") {
+    withTempPath { path =>
+      sql("SELECT 's1' a, 's2' a").write.csv(path.getCanonicalPath)
+      val df = spark.read.csv(path.getCanonicalPath)
+      assert(df.columns === Array("_c0", "_c1"))
```

Comment:
What happens if we want to read csv header as column names? Will it automatically rename duplicate columns (seems this is usually how it is handled, like pandas)?

Reply:
+1 for the above question.

Reply:
The relevant code is https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVUtils.scala#L71-L95

duplicated column names in the header will always be replaced by `_c0`, `_c1`, etc.

Reply:
Got it. Thanks.